src/sentry/api/endpoints/event_file_committers.py

```diff
@@ -30,24 +33,51 @@ def get(self, request: Request, project, event_id) -> Response:
         if event is None:
             return Response({"detail": "Event not found"}, status=404)
 
-        try:
-            committers = get_serialized_event_file_committers(
-                project, event, frame_limit=int(request.GET.get("frameLimit", 25))
+        if features.has("organizations:commit-context", project.organization, actor=request.user):
+            group_owners = GroupOwner.objects.filter(
+                group_id=event.group_id,
+                project=project,
+                organization_id=project.organization_id,
+                type=GroupOwnerType.SUSPECT_COMMIT.value,
+            ).order_by("-date_added")
+            owner = next(filter(lambda go: go.context.get("commitId"), group_owners), None)
+            if not owner:
+                return Response({"committers": []})
+            commit = Commit.objects.get(id=owner.context.get("commitId"))
+
+            return Response(
+                {
+                    "committers": [
+                        {
+                            "author": serialize(owner.user),
+                            "commits": [
+                                serialize(commit, serializer=CommitSerializer(exclude=["author"]))
+                            ],
+                        }
+                    ]
+                }
             )
-        except Group.DoesNotExist:
-            return Response({"detail": "Issue not found"}, status=404)
-        except Release.DoesNotExist:
-            return Response({"detail": "Release not found"}, status=404)
-        except Commit.DoesNotExist:
-            return Response({"detail": "No Commits found for Release"}, status=404)
-
-        data = {
-            "committers": committers,
-        }
-
-        if features.has(
-            "organizations:release-committer-assignees", project.organization, actor=request.user
-        ):
-            data["releaseCommitters"] = get_serialized_release_committers_for_group(event.group)
-
-        return Response(data)
+        else:
+            try:
+                committers = get_serialized_event_file_committers(
+                    project, event, frame_limit=int(request.GET.get("frameLimit", 25))
+                )
+            except Group.DoesNotExist:
+                return Response({"detail": "Issue not found"}, status=404)
```

Comment:
You could use `raise NotFoundError("Issue not found")` from `django_rest_framework.exceptions`. The rest framework exceptions handle setting `detail` and the status code.